webui/src/pages/repositories/repository/objects.jsx

```diff
@@ -226,42 +226,46 @@ const ImportModal = ({config, repoId, referenceId, referenceType, path = '', onD
   );
 };
 
-function extractChecksumFromResponse(response) {
-  if (response.contentMD5) {
-    // convert base64 to hex
-    const raw = atob(response.contentMD5)
-    let result = '';
-    for (let i = 0; i < raw.length; i++) {
-      const hex = raw.charCodeAt(i).toString(16);
-      result += (hex.length === 2 ? hex : '0' + hex);
-    }
-    return result;
-  }
 
-  if (response.etag) {
+
+function extractChecksumFromResponse(parsedHeaders) {
+  if (parsedHeaders['content-md5']) {
     // drop any quote and space
-    return response.etag.replace(/[" ]+/g, "");
+    return parsedHeaders['content-md5'];
   }
-  return ""
+  // fallback to ETag
+  if (parsedHeaders['etag']) {
+    // drop any quote and space
+    return parsedHeaders['etag'].replace(/[" ]+/g, "");
+  }
+  return null;
 }
 
-const uploadFile = async (config, repo, reference, path, file, onProgress) => {
-  const fpath = destinationPath(path, file);
+const uploadFile = async (config, repo, reference, path, file, onProgress) => {  
+  const fpath = destinationPath(path, file);  
   if (config.pre_sign_support_ui) {
-      let additionalHeaders;
-      if (config.blockstore_type === "azure") {
-          additionalHeaders = { "x-ms-blob-type": "BlockBlob" }
-      }
+    let additionalHeaders;
+    if (config.blockstore_type === "azure") {
+      additionalHeaders = { "x-ms-blob-type": "BlockBlob" }
+    }
     const getResp = await staging.get(repo.id, reference.id, fpath, config.pre_sign_support_ui);
-    const uploadResponse = await uploadWithProgress(getResp.presigned_url, file, 'PUT', onProgress, additionalHeaders)
-    if (uploadResponse.status >= 400) {
-      throw new Error(`Error uploading file: HTTP ${status}`)
+    try {
+      const uploadResponse = await uploadWithProgress(getResp.presigned_url, file, 'PUT', onProgress, additionalHeaders);
+      const parsedHeaders = parseRawHeaders(uploadResponse.rawHeaders);
+      const checksum = extractChecksumFromResponse(parsedHeaders);
+      await staging.link(repo.id, reference.id, fpath, getResp, checksum, file.size, file.type);
+    } catch(error) {
+        if (error.status >= 400) {
+          throw new Error(`Error uploading file: HTTP ${error.status}`);
+        }     
+        if (error.status === 0) {
+           throw new Error(`CORS settings error. Check documentation for more info.`);
+        }
```

Comment:
Since there may be a variety of errors, let the error talk for itself.
```suggestion
        throw new Error(`Error uploading file- ${error.status}${error.response ? `: ${error.response}` : ''}`);
```